### ISSUE TITLE
emergency.target does not allow "starting individual units in order to continue the boot process in steps"

### ISSUE THREAD

**Issue by sourcejedi** [systemd/systemd#6509]
### Submission type

 - Bug report

### systemd version the issue has been seen with

233-6.fc26.x86_64 / git HEAD

### In case of bug report: Unexpected behaviour you saw

`man systemd.special` for emergency.target: "allows starting individual units in order to continue the boot
process in steps"

But as soon as you hit systemd-udev-trigger, it will start printer.target (if applicable).  printer.target will pull in cups.  cups will pull in sysinit.target (via DefaultDependencies), and sysinit has `Conflicts=emergency.target emergency.service`.  emergency.service is therefore stopped. 

I have verified that stopping emergency.service kills the emergency shell, and you can't do anything. ctrl+alt+del doesn't work either.

### In case of bug report: Steps to reproduce the problem

1. `systemctl emergency`
2. `systemctl start printer.target` or `systemctl stop emergency.service`

(This was also reproducible on Debian 9 where I am able to boot with `systemd.unit=emergency.target` i.e. #6504 does not apply, unlike Fedora 26 and Fedora 25. Debian 9 has an older version of systemd than Fedora 26 - v232 - which is not specifically supported on this issue tracker.)

**Comment by poettering** [comment]
Yeah, that's a known issue. Starting udev/udev-triggers results in an avalanche of other stuff being pulled in. We should probably comment on that in the documentation in some form.

**Comment by sourcejedi** [comment]
udev-trigger was an example.  I just care that if you try to step to `sysinit.target` as an intermediate, then the emergency console goes away, and you don't get a getty either.

If the situation is known to be that hostile... I guess it would also be useful to mention that udev-trigger could do it indirectly.  But the key point to mention is that if you use this feature, the first thing you need to do is `start debug-shell` as a fallback :).

**Comment by mbiebl** [comment]
Seeing that #6765 was reverted in #6904, I'm going to reopen this bug report.

This issue also popped up downstream at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=991185

**Comment by zackw** [comment]
I'm the original reporter of <https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=991185>.  I would like to mention that on Debian it is normal for package post-install scripts to try to start daemons.  Therefore, if you have a severely crocked system that you're trying to repair, running e.g. `dpkg -a --configure` from the emergency shell might not only lose you the shell, it might interrupt the package manager in the middle of one of these scripts, potentially leaving the system even more broken than before.

**Comment by polarina** [comment]
I've encountered a similar issue:
1. During boot a mount unit fails (required by `initrd-fs.target`) which starts `emergency.target` due to `OnFailure=`.
2. The unit `emergency.target` is started.
3. `sysinit.target` is stopped and finishes stopping due to conflict with `emergency.target`.
4. Right after, by coincidence and unrelated to the failure earlier, `systemd-networkd` contacts D-Bus which triggers `dbus.socket` which starts `dbus.target` which pulls in `sysinit.target` which kills the emergency shell.

As a result, I do not get any emergency shell at all. (Except maybe for a few milliseconds?)
